Sources/NIOHTTP2/HTTP2Frame.swift

```diff
@@ -114,15 +71,86 @@ public struct HTTP2Frame {
         case ping(HTTP2PingData)
         case goAway(lastStreamID: HTTP2StreamID, errorCode: HTTP2ErrorCode, opaqueData: ByteBuffer?)
         case windowUpdate(windowSizeIncrement: Int)
-        case alternativeService
+        case continuation(HTTPHeaders)
+        case alternativeService(origin: String?, field: ByteBuffer?)
+        case blocked
+        case origin([String])
+        case cacheDigest(origin: String?, digest: ByteBuffer?)
+        
+        var code: UInt8 {
+            switch self {
+            case .data:                 return 0x0
+            case .headers:              return 0x1
+            case .priority:             return 0x2
+            case .rstStream:            return 0x3
+            case .settings:             return 0x4
+            case .pushPromise:          return 0x5
+            case .ping:                 return 0x6
+            case .goAway:               return 0x7
+            case .windowUpdate:         return 0x8
+            case .continuation:         return 0x9
+            case .alternativeService:   return 0xa
+            case .blocked:              return 0xb
+            case .origin:               return 0xc
+            case .cacheDigest:          return 0xd
+            }
+        }
+        
+        var allowedFlags: FrameFlags {
+            switch self {
+            case .data:
+                return [.padded, .endStream]
+            case .headers:
+                return [.endStream, .endHeaders, .padded, .priority]
+            case .pushPromise:
+                return [.endHeaders, .padded]
+            case .continuation:
+                return .endHeaders
+            case .cacheDigest:
+                return [.reset, .complete]
+                
+            case .settings, .ping:
+                return .ack
+                
+            case .priority, .rstStream, .goAway, .windowUpdate,
+                 .alternativeService, .blocked, .origin:
+                return []
+            }
+        }
+    }
+    
+    public struct FrameFlags : OptionSet {
+        public typealias RawValue = UInt8
+        
+        public private(set) var rawValue: UInt8
+        
+        public init(rawValue: UInt8) {
+            self.rawValue = rawValue
+        }
+        
+        public static let endStream     = FrameFlags(rawValue: 0x01)
+        public static let ack           = FrameFlags(rawValue: 0x01)
+        public static let reset         = FrameFlags(rawValue: 0x01)
+        public static let complete      = FrameFlags(rawValue: 0x02)
+        public static let endHeaders    = FrameFlags(rawValue: 0x04)
+        public static let padded        = FrameFlags(rawValue: 0x08)
+        public static let priority      = FrameFlags(rawValue: 0x20)
+        
+        // useful for test cases
+        internal static var allFlags: FrameFlags = [.endStream, .endHeaders, .padded, .priority]
```

Comment:
Should this contain `complete`?

Reply:
In theory, yes, but in the implementation it's the same value as `.ack`, so there's not really much value in including it. I could go either way, though.

Reply:
I must be misunderstanding. `.ack` is `0x01` and `.complete` is `0x02`?

Reply:
Er. Yes. Er. Oops. Good catch 😅

Reply:
Le fixed 😊